src/garage/experiment/local_runner.py

```diff
@@ -117,21 +124,28 @@ def obtain_samples(self, itr, batch_size=None):
                 This is a hint that the sampler may or may not respect.
 
         Returns:
-            One batch of samples.
+            list[dict]: One batch of samples.
 
         """
-        if self.train_args.n_epoch_cycles == 1:
+        if self._train_args.n_epoch_cycles == 1:
             logger.log('Obtaining samples...')
-        return self.sampler.obtain_samples(
-            itr, (batch_size or self.train_args.batch_size))
 
-    def save(self, epoch, paths=None):
+        paths = self.sampler.obtain_samples(
+            itr, (batch_size or self._train_args.batch_size))
+
+        for path in paths:
```

Comment:
```python
self._total_env_steps += sum([p['rewards'] for path in path])
```

Reply:
...but perhaps the sampler should just return how many steps it collected?

Reply:
Good idea. So the idea is to change the signature of `obtain_samples`, right? I think I tried this but somehow gave up halfway... I will try it again.

Reply:
it was just an idea. it's a fairly common for APIs which "collect a bunch of stuff" to return "that stuff and how much of it", though that's an old pattern from C.

@krzentner is working on the sampler API so I'm sure she'll want to weigh in on this.

Reply:
I found that changing signature of `sampler.obtain_samples()` is quite intrusive. Because there are many samplers, and I need to use duplicate code to patch each of them.

Instead, I can put it in `runner.obtain_samples()`, which is basically a wrapper of `sampler.obtain_samples()` that meant to do these auxiliary tasks. This is much clearer to me.